ui/common/src/throttle.ts

```diff
@@ -1,56 +1,44 @@
-// Ensures calls to the wrapped function are spaced by the given delay.
-// Any extra calls are dropped, except the last one.
-export default function throttle<T extends (...args: any) => void>(
-  delay: number,
+/***
+ * Wraps an asynchronous function to ensure only one call at a time is in
+ * flight. Any extra calls are dropped, except the last one, which waits for
+ * the previous call to complete.
+ */
+export function throttlePromise<T extends (...args: any) => Promise<void>>(
   callback: T
 ): (...args: Parameters<T>) => void {
-  let timeout: Timeout | undefined;
-  let lastExec = 0;
+  let current: Promise<void> | undefined;
+  let afterCurrent: (() => void) | undefined;
```

Comment:
To review: `afterCurrent` is equivalent to the timeout handle of the old synchronous implementation, `!current` is equivalent to `elapsed > delay`.